src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageFeed.cs

```diff
@@ -25,23 +25,63 @@ public sealed class MultiSourcePackageFeed : IPackageFeed
 
         private readonly SourceRepository[] _sourceRepositories;
         private readonly INuGetUILogger _logger;
+        private readonly INuGetTelemetryService _telemetryService;
 
         public bool IsMultiSource => _sourceRepositories.Length > 1;
 
+        private class TelemetryState
```

Comment:
not sure why do you need this? wouldn't `WaitForCompletionOrBailOutAsync` be completed once per search?

Reply:
Spoke offline.